src/main/java/frc/trigon/robot/utilities/SpeedMechanism2d.java

```diff
@@ -0,0 +1,111 @@
+            currentVelocityBottomArrowLigament,
+            targetVelocityLigament,
+            targetVelocityTopArrowLigament,
+            targetVelocityBottomArrowLigament;
+    private final double deadband;
+
+    public SpeedMechanism2d(String key, double maxDisplayableVelocity) {
+        this(key, maxDisplayableVelocity, 0.001);
+    }
+
+    public SpeedMechanism2d(String key, double maxDisplayableVelocity, double deadband) {
+        this.deadband = deadband;
+        this.key = key;
+        this.mechanism = new Mechanism2d(2 * maxDisplayableVelocity, 2 * maxDisplayableVelocity);
+        MechanismRoot2d root = mechanism.getRoot("VelocityRoot", maxDisplayableVelocity, maxDisplayableVelocity);
+        this.currentVelocityLigament = root.append(new MechanismLigament2d("ZCurrentVelocityLigament", 0, 0, MECHANISM_LINE_WIDTH, new Color8Bit(Color.kBlue)));
+        this.currentVelocityTopArrowLigament = currentVelocityLigament.append(new MechanismLigament2d("ZCurrentVelocityTopArrowLigament", ARROW_LENGTH_SCALE * maxDisplayableVelocity, ZERO_TOP_ANGLE, MECHANISM_LINE_WIDTH, BLUE));
+        this.currentVelocityBottomArrowLigament = currentVelocityLigament.append(new MechanismLigament2d("ZCurrentVelocityBottomArrowLigament", ARROW_LENGTH_SCALE * maxDisplayableVelocity, ZERO_BOTTOM_ANGLE, MECHANISM_LINE_WIDTH, BLUE));
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d("TargetVelocityLigament", 0, 0, MECHANISM_LINE_WIDTH, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityTopArrowLigament", ARROW_LENGTH_SCALE * maxDisplayableVelocity, ZERO_TOP_ANGLE, MECHANISM_LINE_WIDTH, GRAY));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityBottomArrowLigament", ARROW_LENGTH_SCALE * maxDisplayableVelocity, ZERO_BOTTOM_ANGLE, MECHANISM_LINE_WIDTH, GRAY));
+    }
+
+    /**
+     * Updates the mechanism's velocity and target velocity.
+     *
+     * @param velocity       the current velocity
+     * @param targetVelocity the target velocity
+     */
+    public void updateMechanism(double velocity, double targetVelocity) {
+        setTargetVelocity(targetVelocity);
+        updateMechanism(velocity);
+    }
+
+    /**
+     * Updates the mechanism's velocity.
+     * @param velocity the current velocity
+     */
+    public void updateMechanism(double velocity) {
+        setArrowAngle(velocity, currentVelocityTopArrowLigament, currentVelocityBottomArrowLigament);
+        currentVelocityLigament.setLength(velocity);
+        setCurrentLigamentColor(velocityToColor(velocity));
+        Logger.recordOutput(key, mechanism);
+    }
+
+    public void setTargetVelocity(double targetVelocity) {
+        setArrowAngle(targetVelocity, targetVelocityTopArrowLigament, targetVelocityBottomArrowLigament);
+        targetVelocityLigament.setLength(targetVelocity);
```

Comment:
Add a javadoc to this function. Mention in it that this function will not log the mechanism to the network tables